lib/widgets/single_comment.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:flutter/material.dart';
+
+class CommentWidget extends StatefulWidget {
+  @override
+  State<CommentWidget> createState() => _CommentWidgetState();
+  const CommentWidget({super.key});
+}
+
+class _CommentWidgetState extends State<CommentWidget> {
+  int _likes = 0;
+  Color _favIconColor = const Color(0x00000000);
+  Color _favIconOutline = Colors.black;
+
```

Comment:
You can refactor this into a simple _hasLiked boolean to track and change the colors instead of having them as member variables